src/inmanta/execute/scheduler.py

```diff
@@ -232,6 +233,11 @@ def run(self, compiler, statements, blocks):
             xc = ExecutionContext(block, res)
             block.context = xc
             block.namespace.scope = xc
+            block.namespace.warn_shadowed_variables()
+
+        subblocks: Iterator[BasicBlock] = (block for stmt in statements for block in stmt.nested_blocks())
```

Comment:
Please document why this second pass is required. (i.e. why the top level blocks are not enough)